packages/react-client/src/ReactFlightClient.js

```diff
@@ -3393,6 +3410,7 @@ function parseModel<T>(response: Response, json: UninitializedModel): T {
 function createFromJSONCallback(response: Response) {
   // $FlowFixMe[missing-this-annot]
   return function (key: string, value: JSONValue) {
+    isParsingRootModel = key === '';
```

Comment:
We can't assume that an empty string key is the root since it may also be on nested objects. What's the implication of that?

Reply:
Hmm, we were already using the check here:
https://github.com/facebook/react/blob/192555bb0ed88db30f91c58651c421f178f90384/packages/react-client/src/ReactFlightClient.js#L983-L987
But I guess you're right, it could also be an explicit empty key in some object.